Libs/Optimize/ParticleSystem/itkParticleImplicitSurfaceDomain.h

```diff
@@ -71,6 +75,16 @@ class ParticleImplicitSurfaceDomain : public ParticleImageDomainWithCurvature<T,
                                       const PointType &pos) const override;
 
 
+  inline PointType UpdateParticlePosition(PointType &point, vnl_vector_fixed<double, VDimension> &update) const override {
+    PointType newpoint;
+
+    vnl_vector_fixed<float, 3> negativeUpdate;
+    for (unsigned int i = 0; i < 3; i++) { negativeUpdate[i] = -update[i]; }
+    for (unsigned int i = 0; i < 3; i++) { newpoint[i] = point[i] + negativeUpdate[i]; }
```

Comment:
What's the purpose of negativeUpdate?  Why not just subtract the update?

Reply:
@0leks , this is one source of numerical difference between this branch and master.  You have narrowed the update vector from **double** to **float** before the update occurs.  I know the point is only float, but this is affecting the result ever so slightly.

Reply:
This is a remnant from debugging when I was trying to compare image optimization vs mesh optimization

Reply:
Good point on the double to float cast